service/controller/resource/instance/create_cordon_old_workers.go

```diff
@@ -131,22 +134,31 @@ func sortNodesByTenantVMState(nodes []corev1.Node, instances []compute.VirtualMa
 		nodeMap[n.GetName()] = n
 	}
 
+	myVersion := semver.New(project.Version())
+
 	for _, i := range instances {
 		name := instanceNameFunc(customObject, *i.InstanceID)
 
-		if *i.LatestModelApplied {
-			n, found := nodeMap[name]
+		n, found := nodeMap[name]
+		if !found {
 			// When VMSS is scaling up there might be VM instances that haven't
 			// registered as nodes in k8s yet. Hence not all instances are
 			// found from node list.
-			if found {
-				newNodes = append(newNodes, n)
-			}
+			continue
+		}
+
+		v, exists := n.GetLabels()[label.OperatorVersion]
+		if !exists {
+			// Label does not exist, we consider node as old.
```

Comment:
Actually, this might be wrong. I think old nodes always have the operator label (we added it quite a while back) but new nodes don't necessarily have. There's time period when the node has registered but systemd unit that applies labels haven't run yet. You can observe this during upgrade by seeing new node without assigned role in the node list.

Reply:
So what do you suggest to do? Consider the node not new nor old as long as it has the label?
In theory, the azure operator will wait to cordon until he finds the number of old workers equals the number of new workers, so not counting one worker means we would wait for the following reconciliation loop.

Reply:
I made a change to ignore the node from the count if it doesn't have the version label